src/west/commands/project.py

```diff
@@ -38,18 +38,30 @@ def __init__(self):
             '''))
 
     def do_add_parser(self, parser_adder):
-        return _add_parser(parser_adder, self)
+        return _add_parser(parser_adder, self,
+                           _arg('-p', '--path', action='store_true',
+                                help='just print project paths'),
+                           _arg('-a', '--abspath', action='store_true',
+                                help='''just print absolute paths (if given
+                                with -p, this option takes precedence)'''),
+                           _project_list_arg)
 
     def do_run(self, args, user_args):
-        log.inf("Manifest path: {}\n".format(_manifest_path(args)))
-
-        for project in _all_projects(args):
-            log.inf('{:14}  {:18}  {:13}  {}  {}'.format(
-                project.name,
-                project.path,
-                project.revision,
-                project.url,
-                "(cloned)" if _cloned(project) else "(not cloned)"))
+        if not (args.path or args.abspath):
+            log.inf("Manifest path: {}\n".format(_manifest_path(args)))
+
+        for project in _projects(args):
+            if args.abspath:  # This must come before the args.path test.
+                log.inf(project.abspath)
+            elif args.path:
+                log.inf(project.path)
+            else:
```

Comment:
This feels a bit overly specialized to me, just to get rid of some parsing of fixed-format strings.

I wonder if we could have something like this:

    $ west list --format="{name} {path}"

That would print the project name and the relative path (could have `{abspath}` or the like for absolute paths), and would be super simple to implement:

    format_str.format(name=project.name, path=project.path, ...)

The default behavior with no flags could just be a particular format string.

Reply:
No preference here for myself, I don't really mind.

Reply:
No special preference either.
But a small followup question.

Using `--format="{name} {path}"` we get stuff, such as width formatting, for free from built-in python.
But we still need to parse all format names, i.e. `name`, `path`, `abspath`, `...`, or do you have something special lib in mind, which would reduce the parsing we need to do ?

Reply:
@ulfalizer ahh, right.
And `name`, `path`, etc. being set to the internal variable value in project.name, project.path, variables in `west` in `format_str.format(name=project.name, path=project.path, ...)` we only need to keep the local names in the formatting string fixed. Other parts of `west` can change as we find fit.

I actually like that, as it gives freedom to users to fix their formatting and keep it fixed, and even if additional output data/possibilities is provided later.
Such additions will just be silently dropped, if user don't print them.

Used in general throughout west, could gives the script output stability we would like.